js/core/component.js

```diff
@@ -390,23 +390,23 @@ var Component = Class.inherit({
     },
 
     hasActionSubscription: function(actionName) {
-        return !!this._getOptionByStealth(actionName) ||
+        return !!this._optionSilent(actionName) ||
             this._eventsStrategy.hasEvent(this._getEventName(actionName));
     },
 
     isOptionDeprecated(name) {
         return this._options.isDeprecated(name);
     },
 
-    _getOptionByStealth(name) {
-        return this._options.silent(name);
-    },
-
-    _setOptionByStealth(options, value) {
-        this._options.silent(options, value);
+    _optionSilent(options, value) {
+        if(arguments.length < 2 && typeUtils.type(options) !== "object") {
+            return this._options.silent(options);
+        } else {
+            this._options.silent(options, value);
+        }
```

Comment:
Why we check `arguments.length < 2 && typeUtils.type(options) !== "object"` condition twice here and in `options/index.js#169`? Should we do it twice?